src/plugins/kubeconfig/index.ts

```diff
@@ -0,0 +1,138 @@
+/** Copyright © 2024-present P0 Security
+
+This file is part of @p0security/cli
+
+@p0security/cli is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 3 of the License.
+
+@p0security/cli is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with @p0security/cli. If not, see <https://www.gnu.org/licenses/>.
+**/
+import { KubeconfigCommandArgs } from "../../commands/kubeconfig";
+import { waitForProvisioning } from "../../commands/shared";
+import { request } from "../../commands/shared/request";
+import { doc } from "../../drivers/firestore";
+import { print2 } from "../../drivers/stdio";
+import { Authn } from "../../types/identity";
+import { Request } from "../../types/request";
+import { assertNever } from "../../util";
+import { assumeRoleWithIdc } from "../aws/idc";
+import { AwsCredentials } from "../aws/types";
+import { assumeRoleWithOktaSaml } from "../okta/aws";
+import {
+  EksClusterConfig,
+  K8sConfig,
+  K8sGenerated,
+  K8sPermissionSpec,
+} from "./types";
+import { getDoc } from "firebase/firestore";
+import { pick } from "lodash";
+import yargs from "yargs";
+
+export const getAndValidateK8sIntegration = async (
+  authn: Authn,
+  clusterId: string
+): Promise<EksClusterConfig> => {
+  const configDoc = await getDoc<K8sConfig, object>(
+    doc(`o/${authn.identity.org.tenantId}/integrations/k8s`)
+  );
+
+  const clusterConfig = configDoc
+    .data()
+    ?.workflows.items.find(
+      (c) => c.clusterId === clusterId && c.state === "installed"
+    );
```

Comment:
Figuring out if a user can access a cluster involves evaluating the entire routing rule config. This could be a baseline check but can we rely on the backend to evaluate this as part of the `p0 request` command? In that case we don't have to also do it here.

Reply:
Oh, the purpose of this is just to check if a given k8s integration exists; it's not actually evaluating routing rules despite the `workflows` name. This just seems to be how the k8s integrations are stored, in a list under `workflows.items`:
![CleanShot 2024-08-19 at 18 53 26@2x](https://github.com/user-attachments/assets/ae11b388-a807-4c2d-b64f-1803cf70a8d5)